### ISSUE TITLE
Discovery warning on class only when there are other classes

### ISSUE THREAD

**Issue by nohwnd** [microsoft/testfx#2863]
## Describe the bug

[MSTest][Discovery][C:\Users\jajares\source\repos\TestProject97\TestProject97\bin\Debug\net9.0\TestProject97.dll] MSTestAdapter failed to discover tests in class 'TestProject97.ChildWithoutTestClass' of assembly 'C:\Users\jajares\source\repos\TestProject97\TestProject97\bin\Debug\net9.0\TestProject97.dll' because Object reference not set to an instance of an object..

## Steps To Reproduce

The code below shows the warning, but it DOES NOT show the warning, when only `ChildWithoutTestClass` and `ParentWithTestTestClass` are defined. 

The test class ChildWithoutTestClass should NOT find any tests, but also there should be no null ref warning.

[TestProject97.zip](https://github.com/microsoft/testfx/files/15348930/TestProject97.zip)

```powershell
dotnet build;  TestProject97\bin\Debug\net9.0\TestProject97.exe
```

```cs
// file MSTestSettings.cs
[assembly: Parallelize(Scope = ExecutionScope.MethodLevel)]

```

```csproj
<!-- file TestProject97.csproj -->
<Project Sdk="MSTest.Sdk/3.3.1">

  <PropertyGroup>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

  </PropertyGroup>

  <ItemGroup>
    <Using Include="Microsoft.VisualStudio.TestTools.UnitTesting" />
  </ItemGroup>

</Project>

```

```cs
// file WithTestMethods_WithoutTestClass.cs


namespace TestProject97
{
    // SCENARIO1
    // No test class, this loses tests. Would show WARNING.
    // User can either add [TestClass] or mark the class as abstract, if they only intend to inherit the tests.

    public class WithTestMethods_WithoutTestClass
    {
        [TestMethod]
        public void TestMethod1()
        {
        }

        [TestMethod]
        public void TestMethod2()
        {
        }
    }

    // SCENARIO2
    // This test class will inherit parent tests. 2 tests will run from here.
    [TestClass]
    public class WithoutTestMethods_WithTestClass : WithTestMethods_WithoutTestClass
    {

    }

    // This test class will inherit parent tests. Should it show WARNING? IMHO yes.
    public class WithoutTestMethods_WithoutTestClass : WithTestMethods_WithoutTestClass
    {

    }

    // This test class has no tests,
    // it is in examples only to prove that TestClass
    // is not looked up with inheritance (by checking parent class).
    [TestClass]
    public class ParentWithTestTestClass
    {

    }

    // SCENARIO3:
    // Those tests won't run, WARNING should show, the parent class [TestClass] won't be respected, because we don't search
    // inheritance.
    public class ChildWithoutTestClass : ParentWithTestTestClass
    {
        [TestMethod]
        public void TestMethod1()
        {
        }

        [TestMethod]
        public void TestMethod2()
        {
        }
    }

    // SCENARIO4
    // This a valid test class. 2 tests will run from here.
    [TestClass]
    public class UnitTest2
    {
        [TestMethod]
        public void TestMethod1()
        {
        }

        [TestMethod]
        public void TestMethod2()
        {
        }
    }
}

```

## Expected behavior

There is no warning and the tests DON'T run.

**Comment by Evangelink** [comment]
MSTEST0030 is correctly raising in all these cases and I don't see the NRE using 3.5.0. Is the goal of this ticket to produce warnings as part of MSTest execution?

**Comment by nohwnd** [comment]
Re-tested, there is no null reference exception anymore in 3.5.0-preview.24358.4, 

![image](https://github.com/microsoft/testfx/assets/5735905/33f97a1a-fe55-440a-88d0-11a0ec72c339)

the other part of the issue (silently losing tests when [testmethod] is used but no [testclass] is used) should be solved by the new analyzer in 3.6.0 https://github.com/microsoft/testfx/pull/3199 will double check that when reviewing the analyzer article for .net blog.
